extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,49 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
 import { FcUuidAuth } from '../../../../js/common/api/account-servers/flowcrypt-com-api.js';
 import { SmimeKey } from '../../../../js/common/core/crypto/smime/smime-key.js';
 import { PgpHash } from '../../../../js/common/core/crypto/pgp/pgp-hash.js';
+import { UploadedMessageData } from '../../../../js/common/api/account-server.js';
+import { ParsedKeyInfo } from '../../../../js/common/core/crypto/key-store-util.js';
+import { MultipleMessages } from './general-mail-formatter.js';
 
 export class EncryptedMsgMailFormatter extends BaseMailFormatter {
 
-  public sendableMsg = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+  public sendableMsgs = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingKey?: ParsedKeyInfo): Promise<MultipleMessages> => {
     if (newMsg.pwd && !this.isDraft) {
       // password-protected message, temporarily uploaded (already encrypted) to:
       //    - flowcrypt.com/api (consumers and customers without on-prem setup), or
       //    - FlowCrypt Enterprise Server (enterprise customers with on-prem setup)
       //    It will be served to recipient through web
-      const { url: msgUrl, externalId } = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      const uploadedMessageData = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
       newMsg.pwd = undefined;
-      return await this.sendablePwdMsg(newMsg, pubkeys, { msgUrl, externalId }, signingPrv); // encrypted for pubkeys only, pwd ignored
-    } else if (this.richtext) { // rich text: PGP/MIME - https://tools.ietf.org/html/rfc3156#section-4
+      const collectedAttachments = await this.view.attachmentsModule.attachment.collectEncryptAttachments(pubkeys);
+      // we always have at least one pubkey, this is the main message, encrypted for pubkeys
+      const msgs: SendableMsg[] = [
+        await this.sendablePwdMsg(
+          newMsg,
```

Comment:
@tomholub For recipients with a valid pubkey we're currently sending full To/Cc/Bcc headers (including pwd recipients) -- this leads to sort of duplicate messages to pwd recipients (each pwd recipient receives a copy of pubkey-encrypted message with the legacy link and also the pwd-only message without attachments with the personalized link).
If we exclude pwd recipients from the pubkey-encrypted message, then pubkey receivers won't be able to reply-all.
Do you have any particular solution for this in mind?

Reply:
Reply-all from pwd recipient is working ok, but I'm getting this artifact at the bottom of pubkey-encrypted reply. I guess we should file it as a bug?
![image](https://user-images.githubusercontent.com/23335944/166007253-c021be2c-0821-4ba9-8231-a01b2f71b492.png)

Reply:
> @tomholub For recipients with a valid pubkey we're currently sending full To/Cc/Bcc headers (including pwd recipients) -- this leads to sort of duplicate messages to pwd recipients (each pwd recipient receives a copy of pubkey-encrypted message with the legacy link and also the pwd-only message without attachments with the personalized link).
> If we exclude pwd recipients from the pubkey-encrypted message, then pubkey receivers won't be able to reply-all.
> Do you have any particular solution for this in mind?

ping @tomholub

Reply:
not easy

Reply:
To clarify, don't duplicate emails to send them more than once per recipient. The pubkey email would only have pubkey recipients, and other emails only other recipients.

But on the pubkey email, add a `reply-to` header so that the replying client will have other recipients autofilled when they reply. The header accepts a list of recipients.

Reply:
I've been thinking too - without a good solution.

In FES - when I didn't know who was replying (legacy) I set the `Reply-to` to be an amalgamation of `From`, `To` and `Cc` users.  So `To` and `Cc` works.  

Could there be a custom email header `X-reply-all` and put all the emails there?  This assumes all people are using a FlowCrypt extension/plugin though.

Reply:
We could think about custom headers later, once we've done a reasonably working solution with regular headers

Reply:
I tested `reply-to` -- it indeed arrives in a message, but neither GMail nor Flowcrypt picks it up to enable "Reply-All" button. I guess, we should file a separate issue for this? Also (or instead), we can possibly include a text like `This message was also sent to xxx@yyy.com, zzz@xyz.com, don't forget to include them in reply recipients if you'd like to keep them in the loop` etc. This text will be hidden by Flowcrypt in a similar way as the legacy link is currently hidden for pubkey recipients.

Reply:
Ok, let's file this as a separate issue. For me to understand this better:
 - What exactly happens on such message when I click reply?
 - What exactly happens on such message when I click reply all?

Reply:
in Gmail (inside "see original window"), there are 2 possible cases:
1) if the message has 1 pubkey and 1 pwd recipient, https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFqPWLCWgPVsfxGRssLlZQQzn 
![image](https://user-images.githubusercontent.com/23335944/167085709-0a5b4a87-4368-4a19-8158-56690a859c49.png)
GMail doesn't provide "Reply all" button, but pressing "Reply" actually adds 2 recipients (from "Reply-To" header) -- might be a bit misleading for the user.
Flowcrypt doesn't provide "Secure Reply to All" option. pressing "Secure Reply" adds a reply box with empty recipient list.
2) if the message has 2 pubkey and 1 pwd recipient,
https://mail.google.com/mail/u/0/#inbox/FMfcgzGpFgzrRHJVXtgrlxgDgLCZXxGX
Gmail provides "Reply all" button, pressing it adds all 3 recipients to the reply (good).
Flowcrypt doesn't provide "Secure Reply to All" button, pressing "Secure Reply" adds a reply box with empty recipient list.

Reply:
Sounds ok to me for now. In a separate issue, we should ensure that FlowCrypt / secure reply understands reply-to header that has more than one address in it, and fills out the recipients appropriately.